src/unix.rs

```diff
@@ -399,6 +400,80 @@ impl ThreadPriority {
     }
 }
 
+// On apple targets, only `pthread_setschedparam` is allowed. `nice` will fail when increasing the priority if the use is not superuser.
+#[cfg(any(target_os = "macos", target_os = "ios"))]
+impl ThreadPriority {
+    /// Returns the maximum allowed value for using with the provided policy.
+    /// The returned number is in the range of allowed values.
+    pub fn max_value_for_policy(policy: ThreadSchedulePolicy) -> Result<libc::c_int, Error> {
+        let max_priority = unsafe { libc::sched_get_priority_max(policy.to_posix()) };
+        if max_priority < 0 {
+            Err(Error::OS(errno()))
+        } else {
+            Ok(max_priority)
+        }
+    }
+
+    /// Returns the minimum allowed value for using with the provided policy.
+    /// The returned number is in the range of allowed values.
+    pub fn min_value_for_policy(policy: ThreadSchedulePolicy) -> Result<libc::c_int, Error> {
+        let min_priority = unsafe { libc::sched_get_priority_min(policy.to_posix()) };
+        if min_priority < 0 {
+            Err(Error::OS(errno()))
+        } else {
+            Ok(min_priority)
+        }
+    }
+
+    /// Checks that the passed priority value is within the range of allowed values for using with the provided policy.
+    pub fn to_allowed_value_for_policy(
+        priority: libc::c_int,
+        policy: ThreadSchedulePolicy,
+    ) -> Result<libc::c_int, Error> {
+        let min_priority = Self::min_value_for_policy(policy)?;
+        let max_priority = Self::max_value_for_policy(policy)?;
+        let (min, max) = (
+            std::cmp::min(min_priority, max_priority),
+            std::cmp::max(min_priority, max_priority),
+        );
+        let allowed_range = min..=max;
+        if allowed_range.contains(&priority) {
+            Ok(priority)
+        } else {
+            Err(Error::PriorityNotInRange(allowed_range))
+        }
+    }
+
+    /// Converts the priority stored to a posix number.
+    /// POSIX value can not be known without knowing the scheduling policy
+    /// <https://linux.die.net/man/2/sched_get_priority_max>
+    ///
+    /// For threads scheduled under one of the normal scheduling policies (SCHED_OTHER, SCHED_IDLE, SCHED_BATCH), sched_priority is not used in scheduling decisions (it must be specified as 0).
+    /// Source: <https://man7.org/linux/man-pages/man7/sched.7.html>
+    /// Due to this restriction of normal scheduling policies and the intention of the library, the niceness is used
+    /// instead for such processes.
+    pub fn to_posix(self, policy: ThreadSchedulePolicy) -> Result<libc::c_int, Error> {
```

Comment:
Even though there are some minor changes in the code here (it just simplifies the code paths by removing the `Deadline` policy checks for compatible Linux distributions, the code, when executed on a macOS/iOS, should be just the same as in the original implementation., as all the paths not supported on macOS/iOS are ignored.